pulp_puppet_plugins/pulp_puppet/plugins/importers/upload.py

```diff
@@ -38,7 +38,11 @@ def handle_uploaded_unit(repo, type_id, unit_key, metadata, file_path, conduit):
     extracted_data = metadata_parser.extract_metadata(file_path, repo.working_dir)
 
     # rename the file so it has the original module name
-    original_filename = extracted_data['name'] + '-' + extracted_data['version'] + '.tar.gz'
+    # since module's name information is taken from metadata.json, name can be
+    # in forman author/name or author-name. Module's name should be adjusted
+    # before the upload path creation
+    original_filename = extracted_data['name'].replace('/', '-', 1) + '-' + \
+        extracted_data['version'] + '.tar.gz'
```

Comment:
because module's name can differ from the name stored in metadata.json and in the Module itself, as you can see later we override name/author information that is stored in Module https://github.com/ipanova/pulp_puppet/blob/1d68951e12529f2a751defc4cdf01532ed391bd2/pulp_puppet_plugins/pulp_puppet/plugins/importers/upload.py#L50